### PR TITLE
Fix allocator waste assessment in docs

### PR DESCRIPTION
**Pull request by tomusdrw**
Addresses https://github.com/paritytech/substrate/pull/9102/files#r655901183

### CODE REVIEW

**Review by kianenigma — APPROVED** [review]
Could be interesting to add how 25% is deduced as well.

File: primitives/allocator/src/freeing_bump.rs [hunk 1]
```diff
@@ -60,8 +60,8 @@
 //!   fail.
 //!
 //! - Sizes of allocations are rounded up to the nearest order. That is, an allocation of 2,00001 MiB
-//!   will be put into the bucket of 4 MiB. Therefore, typically more than half of the space in allocation
-//!   will be wasted. This is more pronounced with larger allocation sizes.
+//!   will be put into the bucket of 4 MiB. Therefore, on average 25% of the space in allocation
+//!   will be wasted. This is more pronounced in absolute values with larger allocation sizes.
```
Comment (apopiak) [inline primitives/allocator/src/freeing_bump.rs]:
```suggestion
//!   will be wasted. This is more pronounced (in terms of absolute amount of memory wasted) with larger allocation sizes.
```
Is this what you mean?

Comment (tomusdrw) [reply primitives/allocator/src/freeing_bump.rs]: Yes. I've rephrased a bit differently, cause I feel your proposal has too much emphasis on "memory wasted" and I don't think it's just, cause I don't think we can have an ideal allocator that doesn't have any waste at all.

Comment (apopiak) [reply primitives/allocator/src/freeing_bump.rs]: Nice rephrasing!

**Review by apopiak — APPROVED** [review]

**Review by pepyakin — APPROVED** [review]

**Pull request merged**